pkg/gardenadm/botanist/customresourcedefinitions.go

```diff
@@ -0,0 +1,67 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package botanist
+
+import (
+	"context"
+	"fmt"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+
+	"github.com/gardener/gardener/pkg/component/autoscaling/vpa"
+	extensioncrds "github.com/gardener/gardener/pkg/component/extensions/crds"
+	"github.com/gardener/gardener/pkg/component/observability/logging/fluentoperator"
+	"github.com/gardener/gardener/pkg/component/observability/monitoring/prometheusoperator"
+	"github.com/gardener/gardener/pkg/utils/kubernetes/health"
+)
+
+// ReconcileCustomResourceDefinitions reconciles the custom resource definitions.
+func (b *AutonomousBotanist) ReconcileCustomResourceDefinitions(ctx context.Context) error {
+	vpaCRDDeployer := vpa.NewCRD(b.SeedClientSet.Applier(), nil)
+
+	prometheusCRDDeployer, err := prometheusoperator.NewCRDs(b.SeedClientSet.Client(), b.SeedClientSet.Applier())
+	if err != nil {
+		return fmt.Errorf("failed creating Prometheus CRD deployer: %w", err)
+	}
+
+	fluentCRDDeployer, err := fluentoperator.NewCRDs(b.SeedClientSet.Client(), b.SeedClientSet.Applier())
+	if err != nil {
+		return fmt.Errorf("failed creating Prometheus CRD deployer: %w", err)
+	}
+
+	extensionCRDDeployer, err := extensioncrds.NewCRD(b.SeedClientSet.Client(), b.SeedClientSet.Applier(), true, true)
+	if err != nil {
+		return fmt.Errorf("failed creating Prometheus CRD deployer: %w", err)
+	}
+
+	for _, deploy := range []func(context.Context) error{
+		vpaCRDDeployer.Deploy,
+		prometheusCRDDeployer.Deploy,
+		fluentCRDDeployer.Deploy,
+		extensionCRDDeployer.Deploy,
+	} {
+		if err := deploy(ctx); err != nil {
+			return fmt.Errorf("failed to deploy CustomResourceDefinition: %w", err)
```

Comment:
Should we add some context, e.g. the type of the deployer or at least the loop index so that it is clear which one failed in case the error is generic?